x/clp/keeper/keeper_test.go

```diff
@@ -149,3 +151,42 @@ func TestKeeper_GetModuleAccount(t *testing.T) {
 	assert.Equal(t, moduleAccount.GetName(), types.ModuleName)
 	assert.Equal(t, moduleAccount.GetPermissions(), []string{authtypes.Burner, authtypes.Minter})
 }
+
+func TestKeeper_GetLiquidityProviderData(t *testing.T) {
+	ctx, app := test.CreateTestAppClp(false)
+	clpKeeper := app.ClpKeeper
+	tokens := []string{"cada", "cbch", "cbnb", "cbtc", "ceos", "ceth", "ctrx", "cusdt"}
+	pools, lpList := test.GeneratePoolsAndLPs(clpKeeper, ctx, tokens)
+	lpaddr, err := sdk.AccAddressFromBech32(lpList[0].LiquidityProviderAddress)
+	require.NoError(t, err)
+	assetList, _, err := clpKeeper.GetAssetsForLiquidityProviderPaginated(ctx, lpaddr, &query.PageRequest{Limit: math.MaxUint64})
```

Comment:
might be a good idea to iterate over the pages in the test to verify expected behaviour . 
Something like this 
https://github.com/Sifchain/sifnode/blob/0370b7b86b7a6f169731fbbc692e3a0b63cf2c2c/x/dispensation/keeper/distributionRecords_test.go#L132-L144

Reply:
👍